airflow/exceptions.py

```diff
@@ -79,6 +79,12 @@ class DagNotFound(AirflowNotFoundException):
     pass
 
 
+class ExecutionDateAlreadyExists(AirflowBadRequest):
```

Comment:
We may not have to add a new exception class?

Reply:
Yes probably, (and by the way i didn't intend to make a official PR ... I have mistake the offical repo with mine ...

Reply:
should I close the PR and open a new one when I resolve the bugs ?

Reply:
Yep, I would suggest so. Otherwise each of your commits into this branch will trigger a Travis CI build here which will cost Apache foundation some cents ;-)  

Travis CI for your own repo/fork is free though

Reply:
Yes I have already configure it but I just wanted to test the coverage plugin by doing a PR inside my own github (but apparently I was a little tired when I have done it)

I close the PR and I will come back after fixing everything 
Thank !

Reply:
Btw, the way in which you’re using “assertRaises” seems incorrect. That may be partially why your commit failed the CI. You can have a check on that :)